From a0943bd277ed9b5ecaf1c7c1ace7590b1b6da37d Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH] SystemUI: Prevent NPE in Dependency.get() in BrightnessController

When com.android.systemui process died for some reason, the process will
be restarted and will schedule to restart services, such as
SystemUIService and KeyguardService, if start activity BrightnessDialog
before these services, then NPE will happen, the reason is that this
activity will use Dependency which is initalized in these services.

Init Dependency in BrightnessController before using it.

diff --git a/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java b/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
index 66fa000..0114644 100644
--- a/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
+++ b/packages/SystemUI/src/com/android/systemui/settings/BrightnessController.java
@@ -277,6 +277,7 @@ public BrightnessController(Context context, ImageView icon, ToggleSlider contro
         mContext = context;
         mIcon = icon;
         mControl = control;
+        Dependency.initDependencies(context.getApplicationContext());
         mBackgroundHandler = new Handler((Looper) Dependency.get(Dependency.BG_LOOPER));
         mUserTracker = new CurrentUserTracker(mContext) {
             @Override
